controllers/conditions.go

```diff
@@ -23,129 +23,110 @@ import (
 	odfv1alpha1 "github.com/red-hat-data-services/odf-operator/api/v1alpha1"
 )
 
-func SetReconcileInitConditions(conditions *[]conditionsv1.Condition,
-	reason string, message string) {
+func SetAvailableCondition(conditions *[]conditionsv1.Condition,
+	status corev1.ConditionStatus, reason string, message string) {
 
 	conditionsv1.SetStatusCondition(conditions, conditionsv1.Condition{
 		Type:    conditionsv1.ConditionAvailable,
-		Status:  corev1.ConditionFalse,
+		Status:  status,
 		Reason:  reason,
 		Message: message,
 	})
+}
+
+func SetProgressingCondition(conditions *[]conditionsv1.Condition,
+	status corev1.ConditionStatus, reason string, message string) {
+
 	conditionsv1.SetStatusCondition(conditions, conditionsv1.Condition{
 		Type:    conditionsv1.ConditionProgressing,
-		Status:  corev1.ConditionTrue,
+		Status:  status,
 		Reason:  reason,
 		Message: message,
 	})
+}
+
+func SetStorageSystemInvalidCondition(conditions *[]conditionsv1.Condition,
+	status corev1.ConditionStatus, reason string, message string) {
 
 	conditionsv1.SetStatusCondition(conditions, conditionsv1.Condition{
 		Type:    odfv1alpha1.ConditionStorageSystemInvalid,
-		Status:  corev1.ConditionUnknown,
+		Status:  status,
 		Reason:  reason,
 		Message: message,
 	})
+}
+
+func SetVendorCsvReadyCondition(conditions *[]conditionsv1.Condition,
+	status corev1.ConditionStatus, reason string, message string) {
+
 	conditionsv1.SetStatusCondition(conditions, conditionsv1.Condition{
 		Type:    odfv1alpha1.ConditionVendorCsvReady,
-		Status:  corev1.ConditionUnknown,
+		Status:  status,
 		Reason:  reason,
 		Message: message,
 	})
+}
+
+func SetVendorSystemPresentCondition(conditions *[]conditionsv1.Condition,
+	status corev1.ConditionStatus, reason string, message string) {
+
 	conditionsv1.SetStatusCondition(conditions, conditionsv1.Condition{
 		Type:    odfv1alpha1.ConditionVendorSystemPresent,
-		Status:  corev1.ConditionUnknown,
+		Status:  status,
 		Reason:  reason,
 		Message: message,
 	})
 }
 
+func SetReconcileInitConditions(conditions *[]conditionsv1.Condition,
+	reason string, message string) {
+
+	SetAvailableCondition(
+		conditions, corev1.ConditionFalse, reason, message)
+	SetProgressingCondition(
+		conditions, corev1.ConditionTrue, reason, message)
+
+	SetStorageSystemInvalidCondition(
+		conditions, corev1.ConditionUnknown, reason, message)
+	SetVendorCsvReadyCondition(
+		conditions, corev1.ConditionUnknown, reason, message)
+	SetVendorSystemPresentCondition(
+		conditions, corev1.ConditionUnknown, reason, message)
```

Comment:
omg I didn't think about this but this is so much better, thank you!! I would prefer to not have the wrapping, but that's an aesthetic choice. Leave as-is if you like it better.